source/agora/consensus/EnrollmentManager.d

```diff
@@ -309,6 +319,11 @@ public class EnrollmentManager
                 enroll_hash, ex);
             return false;
         }
+
+        // set next height for revealing a pre-image
+        if (enroll_hash == this.enroll_key)
```

Comment:
In the future we should separate these two functionalities of the `EnrollmentManager`:

- Managing enrollments for other nodes
- Managing enrollments for *this* node

Otherwise the implementation will become complex and full of state, like it has started to show.

Reply:
I totally agree with you. I have a plan to split the current `EnrollmentManager` into `EnrollmentManager` and `ValidatorSet`